internal/config/cnimgr/cnimgr.go

```diff
@@ -90,3 +99,31 @@ func (c *CNIManager) Shutdown() {
 		watcher <- false
 	}
 }
+
+// GC calls the plugin's GC to clean up any resources concerned with stale pods
+// (pod other than the ones provided by validPodList). The call to the plugin
+// will be deferred until it is ready logging any errors then and returning nil
+// error here.
+func (c *CNIManager) GC(ctx context.Context, validPodList PodNetworkLister) error {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	if c.lastError != nil {
+		c.validPodList = validPodList
+		return nil
+	}
+	return c.doGC(ctx, validPodList)
+}
+
+func (c *CNIManager) doGC(ctx context.Context, validPodList PodNetworkLister) error {
+	if validPodList != nil {
```

Comment:
nit: this could be inverted and then the rest of the function would be less indented